### PR TITLE
Improve squashing documentation

### PR DESCRIPTION
**Pull request by celestehorgan**
Related to `kubernetes/release` [#1204](https://github.com/kubernetes/release/issues/1204). 

This PR improves documentation and instruction around squashing commits – when you should do it (before a merge), who should do it (the PR author), and what to squash (anything that isn't a meaningful unit of work.)

SIG Docs wants this improvement because we often get contributors who aren't as familiar with Git as some of the other repositories/SIG's. It's difficult to tell a (very) new contributor that they're responsible for squashing their own commits if we don't provide a bit of guidance as to how to do that. Third party resources are good and perhaps preferred, but we often get requests for help with "the end to end process of making my first contribution" regardless of whether we link to third party resources or not. This is a step in that direction.

Similar content will probably land on the `kubernetes/website`'s Contribute section shortly.

### CODE REVIEW

**Comment by celestehorgan** [comment]
@jimangel @justaugustus

**Comment by parispittman** [comment]
I'll let others review for technical but YAAAAAAAAAAAAAS. Thank you @celestehorgan. I've been here a while and am still terrible at this. you rock - thank you!

**Comment by justaugustus** [comment]
Echoing Paris: You rock, @celestehorgan!
At first blush this looks really great. I'll give a full review when I'm back at the laptop later.

/ok-to-test

**Comment by markyjackson-taulia** [comment]
OMG THIS IS AMAZING!
/honk

**Comment by jimangel** [comment]
This is awesome @celestehorgan! Thank you for the quick PR! :tada: 

I think `fixup` vs. `squash` would be situational, but that doesn't impact the effectiveness of your contribution as-is.

/lgtm

File: contributors/guide/github-workflow.md [hunk 1]
```diff
@@ -137,34 +137,101 @@ fork.
...
 
 For mass automated fixups (e.g. automated doc formatting), use one or more
 commits for the changes to tooling and a final commit to apply the fixup en
 masse. This makes reviews easier.
 
+### Merging a commit
+
+Once you've received review and approval, your commits are squashed, your PR is ready for merging. 
+
+Merging happens automatically when a Reviewer or Approver approves the PR. If you haven't squashed your commits, they may ask you to do so before approving a PR. .
```
Comment (mrbobbytables) [inline contributors/guide/github-workflow.md:233]:
```suggestion
Merging happens automatically after both a  Reviewer and Approver have approved the PR. If you haven't squashed your commits, they may ask you to do so before approving a PR.
```

Reviewers cannot approve a PR -- it requires sign off from the approver, but generally is first pass by a reviewer when they sign off with `/lgtm`

File: contributors/guide/pull-requests.md [hunk 2]
```diff
@@ -290,6 +290,8 @@ We might still ask you to clean up your commits at the very end for the sake of
 
 Each commit should have a good title line (<70 characters) and include an additional description paragraph describing in more detail the change intended.
 
+For more information, see [squash commits](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#squash-commits).
```
Comment (mrbobbytables) [inline contributors/guide/pull-requests.md:293]:
```suggestion
For more information, see [squash commits](./github-workflow.md#squash-commits).
```
Nit: This should be a relative link 👍

File: contributors/guide/contributor-cheatsheet/README.md [hunk 3]
```diff
@@ -350,12 +350,20 @@ git checkout -b myfeature
 
 #### Squashing Commits
 
-The main purpose of [squashing commits] is to create a clean readable git
+The main purpose of [squashing commits](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#squash-commits) is to create a clean readable git
```
[the remark this answers is not in the document]
Comment (celestehorgan) [reply contributors/guide/contributor-cheatsheet/README.md:353]: Ahhhhh. I understand now. I saw the square brackets and thought someone forgot to follow through.

[the remark this answers is not in the document]
Comment (mrbobbytables) [reply contributors/guide/contributor-cheatsheet/README.md:353]: No worries :)
Reference style links make it much more readable and can be reused throughout the doc. You then only need to update it once to update every instance of it.

**Comment by mrbobbytables** [comment]
/approve
Thanks!

**Pull request merged**